pkg/virt-controller/watch/vmi/vmi.go

```diff
@@ -1058,14 +1058,14 @@ func (c *Controller) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod, da
 
 		patchedPod, err := c.syncPodAnnotations(pod, newAnnotations)
 		if err != nil {
-			return common.NewSyncError(err, controller.FailedPodPatchReason)
+			return common.NewSyncError(err, controller.FailedPodPatchReason), pod
 		}
-		*pod = *patchedPod
+		pod = patchedPod
```

Comment:
You do fix the corruption but this is all very subtle and fragile, do we have a unit test that checks the original pod object doesn't get corrupted? My point is that even if you now return a different pod object it's likely that someone else modifies `sync` and corrupts it again.

This is not to block this PR but I think we should think of a better future-proof way to avoid making this mistake again.